common/src/main/kotlin/io/homeassistant/companion/android/util/UrlUtil.kt

```diff
@@ -168,3 +168,38 @@ private fun InetAddress.isPrivateOrLocal(): Boolean {
         // 169.254.0.0/16 or fe80::/10
         this.isAnyLocalAddress // 0.0.0.0 or ::
 }
+
+/**
+ * Checks if this URL has the same origin (scheme, host, and port) as the other URL.
+ *
+ * @param other the URL to compare against
+ * @return `true` if both URLs have the same scheme, host, and port
+ */
+fun HttpUrl.hasSameOrigin(other: HttpUrl): Boolean {
+    return scheme == other.scheme &&
+        host == other.host &&
+        port == other.port
+}
+
+/**
+ * Checks if this Uri has the same base (scheme, host, and port) as the other Uri.
+ *
+ * @param other the Uri to compare against
+ * @return `true` if both URIs have the same scheme, host, and port
+ */
+fun Uri.hasSameBase(other: Uri?): Boolean {
+    if (other == null) return false
+    return scheme.equals(other.scheme, ignoreCase = true) &&
+        host.equals(other.host, ignoreCase = true) &&
+        port == other.port
+}
+
+/**
+ * Checks if this Uri has a meaningful path (not empty, not just "/").
+ *
+ * @return `true` if the Uri has a path that is not blank and not just "/"
+ */
+fun Uri.hasMeaningfulPath(): Boolean {
```

Comment:
You need to check the KDoc here to understand what is "meaningful". I don't really like it and would prefer something more intuitive like Kotlins `isNullOrBlank`, but `Uri.pathIsNullOrBlankOrSlash()` doesn't quite have the same ring to it :)